bundles/dna_templates/agent/dna.yaml

```diff
@@ -1,34 +1,50 @@
-  remote_auth:
-    permissions:
-      - extern_id: index_agent_commitments_as_provider
-        allowed_method: [agent_index, index_agent_commitments_as_provider]
-      - extern_id: index_agent_commitments_as_receiver
-        allowed_method: [agent_index, index_agent_commitments_as_receiver]
-      - extern_id: index_agent_intents_as_provider
-        allowed_method: [agent_index, index_agent_intents_as_provider]
-      - extern_id: index_agent_intents_as_receiver
-        allowed_method: [agent_index, index_agent_intents_as_receiver]
-      - extern_id: index_agent_economic_events_as_provider
-        allowed_method: [agent_index, index_agent_economic_events_as_provider]
-      - extern_id: index_agent_economic_events_as_receiver
-        allowed_method: [agent_index, index_agent_economic_events_as_receiver]
-      - extern_id: index_agent_inventoried_economic_resources
-        allowed_method: [agent_index, index_agent_inventoried_economic_resources]
-zomes:
-  - name: agent
-    path: "<repository-path>/target/wasm32-unknown-unknown/release/hc_zome_rea_agent.wasm"
-  - name: agent_index
-    path: "<repository-path>/target/wasm32-unknown-unknown/release/hc_zome_rea_agent_index_agent.wasm"
+integrity:
+  uid: ""
+  origin_time: 2022-07-01T00:00:00.000000Z
+  properties:
+    time_index_chunk_interval_ms: 30000
+    agent:
+      index_zome: agent_index
+    agent_index:
+      record_storage_zome: agent
+    remote_auth:
+      permissions:
+        - extern_id: index_agent_commitments_as_provider
+          allowed_method: [agent_index, index_agent_commitments_as_provider]
+        - extern_id: index_agent_commitments_as_receiver
+          allowed_method: [agent_index, index_agent_commitments_as_receiver]
+        - extern_id: index_agent_intents_as_provider
+          allowed_method: [agent_index, index_agent_intents_as_provider]
+        - extern_id: index_agent_intents_as_receiver
+          allowed_method: [agent_index, index_agent_intents_as_receiver]
+        - extern_id: index_agent_economic_events_as_provider
+          allowed_method: [agent_index, index_agent_economic_events_as_provider]
+        - extern_id: index_agent_economic_events_as_receiver
+          allowed_method: [agent_index, index_agent_economic_events_as_receiver]
+        - extern_id: index_agent_inventoried_economic_resources
+          allowed_method: [agent_index, index_agent_inventoried_economic_resources]
+  zomes:
+    # application zomes
+    - name: agent_integrity
+      bundled: "<repository-path>/target/wasm32-unknown-unknown/release/hc_zome_rea_agent_integrity.wasm"
+
+    # utility zomes
+    - name: remote_auth_integrity
+      bundled: "<repository-path>/target/wasm32-unknown-unknown/release/hc_zome_dna_auth_resolver_hrea_integrity.wasm"
+
+coordinator:
+  zomes:
+    # application zomes
+    - name: agent
+      bundled: "<repository-path>/target/wasm32-unknown-unknown/release/hc_zome_rea_agent.wasm"
+    - name: agent_index
```

Comment:
For all of this to work, there might need to be an 'index_integrity' zome as well, that gets added as a dependency to all the index zomes. It might be able to be the same zome for all of them

Reply:
are there any new integrity types that would need to get defined though? Can we not just import the `LinkTypes` from the same integrity zome?

Reply:
I guess I was thinking we only need to define 1 integrity zome per dna, but if we are to split every zome into coordinator and integrity then we'd have 2

Reply:
Ah, I see. that could be true. Depends if the index zomes need entry types, or just link types. And if we were to make LinkTypes more specific then we'd need more specific integrity zomes. 

Your proposal could work